include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,648 @@
+
+        @param other string to compare
+
+        @return Negative value if this string is less than the other
+        character sequence, zero if the both character sequences are
+        equal, positive value if this string is greater than the other
+        character sequence
+    */
+    BOOST_URL_DECL
+    int
+    compare(decoded_view other) const noexcept;
+
+    /** Returns a string with the decoded data
+    */
+    std::string
+    to_string() const
+    {
+        std::string r;
+        assign_to(r);
+        return r;
+    }
+
+    /** Assigns the decoded characters to a container
+
+        This overload allows the caller to recycle
+        the input string.
+
+        This function only participates in
+        overload resolution if the container
+        type `T` has the member function
+        `assign(iterator, iterator)`.
+
+        @return A string representing the
+        entire contents of the decoded range.
+    */
+    template <class String>
+    String&
+    assign_to(String& s) const;
+
+    /** Appends the decoded characters to a container
+
+        This overload allows the caller to recycle
+        the input string.
+
+        This function only participates in
+        overload resolution if the container
+        type `T` has the member function
+        `append(iterator, iterator)`.
+
+        @return A string representing the
+        entire contents of the decoded range.
+    */
+    template <class String>
+    String&
+    append_to(String& s) const;
+
+    /** Return true if two strings are equal
+    */
+    friend
+    bool
+    operator==(
+        decoded_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return lhs.compare(rhs) == 0;
+    }
+
+    /** Return true if two strings are unequal
+    */
+    friend
+    bool
+    operator!=(
+        decoded_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return lhs.compare(rhs) != 0;
+    }
+
+    /** Return true if lhs is less than rhs
+    */
+    friend
+    bool
+    operator<(
+        decoded_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return lhs.compare(rhs) < 0;
+    }
+
+    /** Return true if lhs is less or equal to rhs
+    */
+    friend
+    bool
+    operator<=(
+        decoded_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return lhs.compare(rhs) <= 0;
+    }
+
+    /** Return true if lhs is greater than rhs
+    */
+    friend
+    bool
+    operator>(
+        decoded_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return lhs.compare(rhs) > 0;
+    }
+
+    /** Return true if lhs is greater or equal to rhs
+    */
+    friend
+    bool
+    operator>=(
+        decoded_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return lhs.compare(rhs) >= 0;
+    }
+
+    /** Return true if two strings are equal
+
+        @note This function only participates in
+        overload resolution if when a `string_view`
+        is constructible from a `String`.
```

Comment:
Don't we use `@par Constraints`?

```
        @par Constraints
        @code
        std::is_convertible_v< String, string_view >
        @endcode
```

Check out this javadoc, it has a little bit of everything:
https://github.com/boostorg/json/blob/43d04d1d214e29d1ad890c5f9237ae0201726166/include/boost/json/array.hpp#L1241